djstripe/views.py

```diff
@@ -189,18 +189,17 @@ def post(self, request, *args, **kwargs):
                 # we force the proration of the current plan and use it towards the upgraded plan,
                 # no matter what DJSTRIPE_PRORATION_POLICY is set to.
                 if PRORATION_POLICY_FOR_UPGRADES:
+                    # TODO: Needs refactor
                     current_subscription_amount = customer.current_subscription.amount
-                    selected_plan_name = form.cleaned_data["plan"]
-                    selected_plan = next(
-                        (plan for plan in PLAN_LIST if plan["plan"] == selected_plan_name)
-                    )
-                    selected_plan_price = selected_plan["price"] / decimal.Decimal("100")
+                    selected_plan_id = form.cleaned_data["plan"]
+                    selected_plan = Plan.objects.get(stripe_id=selected_plan_id)
+                    selected_plan_price = selected_plan.amount / decimal.Decimal("100")
```

Comment:
Is the decimal division necessary here? I think amount is already a decimal in the model.